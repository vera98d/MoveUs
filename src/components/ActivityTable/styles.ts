import styled from "styled-components";

export const TableWrapper = styled.div`
  width: 80vh;
  height: 70vh;
  display: flex;
  flex-direction: column;
  align-items: center;
`;

export const ExerciseTable = styled.div`
  display: grid;
  text-align: center;
  grid-template-columns: repeat(4, 1fr);
  grid-template-rows: [row1-start] 15%;
  column-rule: 1px solid red;
  grid-auto-rows: 12.14%;
  border: 1px solid black;
  width: 100%;
  height: 50vh;
`;

export const ExerciseCell = styled.p`
  padding: 8px 4px;
  border-left: 1px solid black;
  border-bottom: 1px solid black;
`;

export const PaginationWrapper = styled.div`
  width: 80%;
  display: flex;
  justify-content: space-around;
`;

export const PaginationList = styled.ul`
  display: flex;
  list-style-type: none;
  
  .linkStyle {
    text-decoration: none;
    color: inherit;
  }
`;

export const ListElement = styled.li`
`;
