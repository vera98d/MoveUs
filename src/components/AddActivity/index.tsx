import { SubmitHandler, useForm } from "react-hook-form";
import {
  AccountText,
  Button, Container, Form, FormField, FormFieldError, Label, Img, Input, StyledLink, Wrapper,
} from "../Form/styles";
import { Activity } from "../../interfaces/dbData";
import authService from "../../services/authService";
import { useAuthState } from "react-firebase-hooks/auth";
import activityService from "../../services/activityService";

type ActivityField = Omit<Activity, "id" | "score">;

function AddActivity() {
  const [user] = useAuthState(authService.getAuth());
  const { register, handleSubmit, formState: { errors } } = useForm<ActivityField>();
  const onSubmit: SubmitHandler<ActivityField> = (data) => activityService
    .insert(data, user!.uid);
  const onSubmitError: SubmitHandler<any> = (data) => console.log(data, errors);

  const exercises: string[] = ["running", "cycling", "walking"];
  // TODO get these from DB

  return (
    <Wrapper>
      <Form onSubmit={handleSubmit(onSubmit, onSubmitError)}>
        <FormField>
          <Label>Select exercise.</Label>
          <select {...register("exercise", { required: true, minLength: 3, maxLength: 20 })}>
            { exercises.map((e) => {
              return <option value={e}>{e}</option>;
            })}
          </select>
          <FormFieldError>
            {errors.exercise?.type === "required" && "Please select  the activity type"}
          </FormFieldError>
        </FormField>
        <FormField>
          <Label>Enter date of the activity.</Label>
          <input
            type="date"
            {...register("date", { required: true })}
          />
          <FormFieldError>
            {errors.date?.type === "required" && "Please select date"}
          </FormFieldError>
        </FormField>
        <FormField>
          <Label>Select duration of the exercise..</Label>
          <input
            type="time"
            {...register("duration", { required: true })}
          />
          <FormFieldError>
            {errors.exercise?.type === "required" && "Please select  the activity type"}
          </FormFieldError>
        </FormField>
        <Button>Login</Button>
      </Form>
    </Wrapper>
  );
}

export default AddActivity;
