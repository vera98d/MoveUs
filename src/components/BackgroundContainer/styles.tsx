import styled from "styled-components";

const BackgroundContainer = styled.div`
    width: 100%;
    min-height: 100vh;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    padding: 2em 3em;
    background: ${(props) => props.theme.colors.background};
`;

export default BackgroundContainer;