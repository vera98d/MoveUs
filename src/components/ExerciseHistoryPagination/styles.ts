import styled from "styled-components";

export const PaginationList = styled.ul`
  display: flex;
  list-style-type: none;
  
  .linkStyle {
    text-decoration: none;
    color: inherit;
  }
`;

export const ListElement = styled.li`
`;
