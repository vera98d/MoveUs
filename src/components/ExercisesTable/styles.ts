import styled from "styled-components";

interface EmptyGridLineProps {
  lines: number;
}

export const PaginationWrapper = styled.div`
  margin: 20px 0;
  width: 100%;
  max-width: 1050px;
  display: flex;
  justify-content: center;
`;

export const PaginationList = styled.ul`
  display: flex;
  list-style-type: none;

  .linkStyle {
    text-decoration: none;
    color: inherit;
  }
`;

export const ScoreLine = styled.div`
  margin: 20px 0;
  width: 100%;
  max-width: 1050px;
  display: flex;
  justify-content: space-between;
  align-items: stretch;
  position: relative;

  .buttonFontSize {
    font-size: 16px;
  }

  .sumPosition {
    position: absolute;
    left: 66%;
    font-family: ${(props) => props.theme.fontFamily.primaryFont};
    font-size: 16px;
    color: ${(props) => props.theme.colors.secondaryUi};
  }
`;

export const PageNumber = styled.li``;

export const GridContainer = styled.div`
  width: 100%;
  max-width: 1050px;
  border-radius: 15px;
  border: ${(props) => props.theme.border};
  box-shadow: ${(props) => props.theme.boxShadow};
  -moz-box-shadow: ${(props) => props.theme.boxShadow};
  -webkit-box-shadow: ${(props) => props.theme.boxShadow};
  background: ${(props) => props.theme.colors.white};
`;

export const GridLine = styled.div`
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-template-rows: [row1-start] 15%;
  border-bottom: ${(props) => props.theme.border};

  &:nth-last-child(1) {
    border-bottom: none;
  }
`;

export const GridChild = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  flex-wrap: wrap;
  padding: 0.5em 1em;
  color: ${(props) => props.theme.colors.secondaryUi};
  font-family: ${(props) => props.theme.fontFamily.primaryFont};
  font-size: 12px;
  border-right: ${(props) => props.theme.border};
  min-height: 35px;
  &:nth-child(4n) {
    border-right: none;
  }
`;

export const GridHeader = styled(GridChild)`
  color: ${(props) => props.theme.colors.secondaryUi};
  font-size: 14px;
  min-height: 50px;
`;

export const BlankLine = styled.div`
  display: grid;
  grid-template-columns: repeat(2, 1fr);
`;

export const EmptyGridLine = styled(BlankLine)<EmptyGridLineProps>`
  border-bottom: none;
  min-height: ${(props) =>
    props.lines !== 0 ? `${props.lines * 35}px` : "0px"};
`;

export const SiteSetterWrapper = styled.div`
  display: flex;
  justify-content: center;
`;

export const PageButton = styled.button`
  margin: 0 10px;
  border: none;
  color: ${(props) => props.theme.colors.secondaryUi};
  font-family: ${(props) => props.theme.fontFamily.primaryFont};
  font-weight: 900;
  font-size: 18px;
  cursor: pointer;
  background-color: transparent;
`;
