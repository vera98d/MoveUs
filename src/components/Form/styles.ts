import { Link } from "react-router-dom";
import styled from "styled-components";

export const Container = styled.div`
background-image: url("${process.env.PUBLIC_URL}/assets/background_photo.jpg");
background-repeat: no-repeat;
background-size: cover;
background-position: 0% 90%;
height: 100vh;
`;

export const Img = styled.img`
position: absolute;
height: 100px;
`;

export const Wrapper = styled.div`
display: flex;
justify-content: flex-end;
height: 100%;
max-width: 1320px;
margin: 0 auto;
padding: 0 10px;
`;

export const Form = styled.form`
display: flex;
flex-direction: column;
justify-content: center;
align-items: center;
align-content: center;
gap: 20px;
align-items: stretch;
 `;

export const FormField = styled.div`
display: flex;
flex-direction: column;
gap: 5px;
`;

export const Label = styled.p`
color: ${(props) => props.theme.colors.white};
font-family: ${(props) => props.theme.fontFamily.primaryFont};
font-size: 15px;
`;

export const Input = styled.input`
display: flex;
text-align: left;
padding: 1.2em;
border: solid 2px ${(props) => props.theme.colors.secondaryUi};
border-radius: 15px;
background-color: ${(props) => props.theme.colors.primaryUi};
font-family: ${(props) => props.theme.fontFamily.primaryFont};
font-size: 15px;
color: ${(props) => props.theme.colors.secondaryUi};
width: 100%;
max-width: 400px;

  &:focus {
    outline: none;
    box-shadow: 0px 0px 10px ${(props) => props.theme.colors.secondaryUi};
  }
`;

export const Button = styled.button`
display: flex;
  align-items: center;
  justify-content: center;
  align-self: center;
  background-color: ${(props) => props.theme.colors.secondaryUi};
  color: ${(props) => props.theme.colors.primaryUi};
  font-family: ${(props) => props.theme.fontFamily.primaryFont};
  font-size: 18px;
  padding: 0.75em 4em;
  border-radius: 10px;
  border: none;
  cursor: pointer;
  margin-top: 20px;
  margin-bottom: 30px;

   &:hover{
    box-shadow: 0px 0px 10px ${(props) => props.theme.colors.secondaryUi};
  }
`;

export const FormFieldError = styled.span`
text-align: center;
color: ${(props) => props.theme.colors.secondaryUi};
font-family: ${(props) => props.theme.fontFamily.primaryFont};
font-size: 15px;
`;

export const AccountText = styled.div`
color: ${(props) => props.theme.colors.white};
font-family: ${(props) => props.theme.fontFamily.primaryFont};
font-size: 18px;
align-self: center;
`;

export const StyledLink = styled(Link)`
text-decoration: none;
color:${(props) => props.theme.colors.white};
font-weight: bold;

  &:hover{
    color: ${(props) => props.theme.colors.secondaryUi};
  }
`;
