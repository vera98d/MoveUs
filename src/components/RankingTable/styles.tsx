import styled from "styled-components";

export const GridContainer = styled.div`
    width: 90%;
    max-width: 1050px;
    border-radius: 15px;
    border: ${(props) => props.theme.border};
    box-shadow: ${(props) => props.theme.boxShadow};
    -moz-box-shadow: ${(props) => props.theme.boxShadow};
    -webkit-box-shadow: ${(props) => props.theme.boxShadow};
    background: ${(props) => props.theme.colors.white};
`;

export const GridLine = styled.div`
    display: grid;
    grid-template-columns: 1fr 1.6fr 1fr 1.3fr;
    border-bottom: ${(props) => props.theme.border};

    &:nth-last-child(1){
        border-bottom: none;
    }
`;

export const GridChild = styled.div`
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
    flex-wrap: wrap;
    padding: .5em 1em;
    color: ${(props) => props.theme.colors.secondaryUi};
    font-family: ${(props) => props.theme.fontFamily.primaryFont};
    font-size: 18px;
    border-right: ${(props) => props.theme.border};
   
    &:nth-child(4n){
        border-right: none;
    }
`;

export const GridHeader = styled(GridChild)`
    color: ${(props) => props.theme.colors.secondaryUi};
    font-size: 22px;
`;
