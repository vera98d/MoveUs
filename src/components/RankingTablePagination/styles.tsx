import styled from "styled-components";

interface NavigationButtonProps {
  enabled: boolean;
}

export const NavContainer = styled.div`
  display: flex;
  width: 25%;
  gap: 1em;
  justify-content: center;
  margin-top: 2em;
`;

export const CurrentPageNumber = styled.h2`
  color: ${(props) => props.theme.colors.secondaryUi};
  font-size: ${(props) => props.theme.fontSize.text};
`;

export const NavigationButton = styled.div<NavigationButtonProps>`
  display: block;
  cursor: ${(props) => (props.enabled ? "pointer" : "default")};
  color: ${(props) =>
    props.enabled ? props.theme.colors.secondaryUi : "#7f8599"};
  font-size: ${(props) => props.theme.fontSize.text};
  font-weight: bold;
  transition: 0.5s;
`;
