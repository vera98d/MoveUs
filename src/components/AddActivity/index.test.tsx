import { render, screen, waitFor, fireEvent } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import React from "react";
import { ThemeProvider } from "styled-components";
import { theme } from "../../styles";
import AddActivity from ".";

test("is rendering expected elements", async () => {
  const { container } = render(
    <ThemeProvider theme={theme}>
      <AddActivity />
    </ThemeProvider>,
  );

  expect(container).toMatchSnapshot();
});

test("Renders all elements", () => {
  render(
    <ThemeProvider theme={theme}>
      <AddActivity />
    </ThemeProvider>,
  );
  const modalTitle = screen.getByText(/Add/i);
  const selectExerciseText = screen.getByText(/Select exercise/i);
  const selectExerciseInput = screen.getByRole("combobox");
  const enterDateText = screen.getByText(/Enter date of the activity/i);
  const selectDateInput = screen.getByTestId("date");
  const enterDurationText = screen.getByText(/Select duration of the exercise/i);
  const selectDurationInput = screen.getByTestId("time");
  const pointsText = screen.getByText(/Points gained:/i);
  const pointsEdit = screen.getByTestId("score");
  const submitButton = screen.getByText("Submit activity");

  expect(modalTitle).toBeInTheDocument();

  expect(selectExerciseInput).toBeInTheDocument();
  expect(selectExerciseText).toBeInTheDocument();

  expect(enterDateText).toBeInTheDocument();
  expect(selectDateInput).toBeInTheDocument();

  expect(enterDurationText).toBeInTheDocument();
  expect(selectDurationInput).toBeInTheDocument();

  expect(pointsText).toBeInTheDocument();
  expect(pointsEdit).toBeInTheDocument();

  expect(submitButton).toBeInTheDocument();
});

test("validates form properly", async () => {
  render(
    <ThemeProvider theme={theme}>
      <AddActivity />
    </ThemeProvider>,
  );
  const submitButton = screen.getByText("Submit activity");
  const selectDateInput = screen.getByTestId("date") as HTMLInputElement;
  const selectDurationInput = screen.getByTestId("time") as HTMLInputElement;

  userEvent.click(submitButton);

  await waitFor(() => {
    expect(screen.getByText("Please select the activity type.")).toBeInTheDocument();
    expect(screen.getByText("Please select date.")).toBeInTheDocument();
    expect(screen.getByText("Please enter duration of your activity.")).toBeInTheDocument();
  });
  userEvent.type(selectDateInput, "2005-04-02");
  userEvent.type(selectDurationInput, "21:37");

  userEvent.click(submitButton);

  await waitFor(() => {
    expect(screen.queryByText("Please select date.")).not.toBeInTheDocument();
    expect(screen.queryByText("Please enter duration of your activity.")).not.toBeInTheDocument();
  });
});