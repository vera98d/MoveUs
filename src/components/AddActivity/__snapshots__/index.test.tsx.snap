// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`is rendering expected elements 1`] = `
<div>
  <h1
    class="sc-hKMtZM jdDlkz"
  >
    Add activity
  </h1>
  <form
    class="sc-papXJ eFThpA"
  >
    <div
      class="sc-jqUVSM dFGdLF"
    >
      <p
        class="sc-jSMfEi iUXMgt"
      >
        Select exercise
      </p>
      <select
        name="exercise"
      >
        <option
          value=""
        >
           
        </option>
      </select>
      <span
        class="sc-evZas iBuEyh"
      />
    </div>
    <div
      class="sc-jqUVSM dFGdLF"
    >
      <p
        class="sc-jSMfEi iUXMgt"
      >
        Enter date of the activity
      </p>
      <input
        data-testid="date"
        name="date"
        type="date"
      />
      <span
        class="sc-evZas iBuEyh"
      />
    </div>
    <div
      class="sc-jqUVSM dFGdLF"
    >
      <p
        class="sc-jSMfEi iUXMgt"
      >
        Select duration of the exercise
      </p>
      <input
        data-testid="time"
        name="duration"
        type="time"
      />
      <span
        class="sc-evZas iBuEyh"
      />
    </div>
    <p
      class="sc-jSMfEi iUXMgt"
    >
      Points gained:
    </p>
    <div
      class="sc-jqUVSM dFGdLF"
    >
      <input
        data-testid="score"
        name="score"
        readonly=""
      />
    </div>
    <button
      class="sc-crXcEl iThiiz"
    >
      Submit activity
    </button>
  </form>
</div>
`;
